src/device_trezor/trezor/transport.hpp

```diff
@@ -84,6 +86,15 @@ namespace trezor {
     additional_params.push_back(std::make_pair("Content-Type","application/json; charset=utf-8"));
 
     const http::http_response_info* pri = nullptr;
+    const auto data_cleaner = epee::misc_utils::create_scope_leave_handler([&]() {
+      if (!req_param.empty()) {
+        memwipe(&req_param[0], req_param.size());
+      }
+      if (pri != nullptr && !pri->m_body.empty()){
+        memwipe(&(const_cast<http::http_response_info*>(pri)->m_body)[0], pri->m_body.size());
```

Comment:
That's really iffy. Looking at the code, maybe you could add a transport.wipe_response, which would wipe.

Reply:
I've added `transport.wipe_response`, pls check if that is what you had in mind

Reply:
No. This change just moves the code, but does the same. What I had in mind is adding a "wipe" function in http_client (or whatever class t_transport is here), then call wipe  instead of calling memwipe. This avoids the const cast.

Reply:
Aah so direct modification of the simple_http_client, OK I can do that

Reply:
@moneromooo-monero pls check if this is what you had in mind `transport.wipe_response();`. I've called it `wipe_response` as the client has another fields that could be cleaned in some scenarios, such as http credentials, but we are not using it right now.

- https://github.com/monero-project/monero/pull/6446/files#diff-f53fddf8049bb20838aa3e7698a69509
- https://github.com/monero-project/monero/pull/6446/files#diff-dff12913bf1880e90814097d1eedbcdd
- https://github.com/monero-project/monero/pull/6446/files#diff-fbd330073222eed440fa85924896bcd3

Reply:
Yes, that's exactly what I intended. Thanks.